pkg/R/DataFrame.R

```diff
@@ -0,0 +1,119 @@
+# DataFrame.R - DataFrame class and methods implemented in S4 OO classes
+
+#' @include jobj.R RDD.R
+NULL
+
+setOldClass("jobj")
+
+#' @title S4 class that represents a DataFrame
+#' @description DataFrames can be created using functions like
+#'              \code{jsonFile}, \code{table} etc.
+#' @rdname DataFrame
+#' @seealso jsonFile, table
+#'
+#' @param env An R environment that stores bookkeeping states of the DataFrame
+#' @param sdf A Java object reference to the backing Scala SchemaRDD
+#' @export
+
+setClass("DataFrame",
+         slots = list(env = "environment",
+                      sdf = "jobj"))
+
+setMethod("initialize", "DataFrame", function(.Object, sdf, isCached, isCheckpointed) {
+  .Object@env <- new.env()
+  .Object@env$isCached <- isCached
+  .Object@env$isCheckpointed <- isCheckpointed
+  
+  .Object@sdf <- sdf
+  .Object
+})
+
+#' @rdname DataFrame
+#' @export
+
+dataFrame <- function(sdf, isCached = FALSE, isCheckpointed = FALSE) {
+  new("DataFrame", sdf, isCached, isCheckpointed)
+}
+
+# The DataFrame accessor function
+
+setGeneric("getsdf", function(df, ...) {standardGeneric("getsdf") })
```

Comment:
getsdf is here mostly as a placeholder in case I need to add an additional method to it later.  I wasn't sure if I'd need to extend the DataFrame class later (like you did with PipelinedRDD) so I built the option in just in case I need to add to it later. 

If you think we might need it later, I'd prefer to leave it in during development so that later refactors will be easier.  That said, I haven't seen any cases in the 1.3 source where there's another class that extends `DataFrame.scala` so it maybe worth just taking out now.

What do you think?